minlzlib/xzstream.c

```diff
@@ -463,18 +484,24 @@ XzDecode (
     //
     // Decode the block header for check for validity
     //
-    if (!XzDecodeBlockHeader())
+    switch (XzDecodeBlockHeader())
     {
+    case XzBlockHeaderFail:
         return false;
+    case XzBlockHeaderNoBlock:
+        *OutputSize = 0;
```

Comment:
I understand that the intent of the 'error value' was to know that we don't have to decode a block. I'm trying to think of a different way to achieve this...

Reply:
Yeah, it was more of a "multiple success codes" situation rather than a "multiple error vales" one.

The way I see it, we have two options:
1) XzDecodeBlockHeader takes an additional `*bool` parameter and fills it with "do we have a block to process"
2) XzDecodeBlockHeader returns something other than a bool - a `uint8_t` with `#define` d constants, an enum, etc.

Reply:
I suppose there's also the third option of "Leave compressed empty files as a known failure case"